community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelStatement.java

```diff
@@ -185,5 +198,103 @@ private void cleanupResources()
         }
 
         transaction.releaseStatement( this );
+
+        if ( storeStatement != null )
+        {
+            storeStatement.close();
+        }
+    }
+
+    public TxNodeCursor acquireNodeCursor()
+    {
+        if ( nodeCursor == null )
+        {
+            nodeCursor = new TxNodeCursor( this );
+        }
+
+        try
+        {
+            return nodeCursor;
+        }
+        finally
+        {
+            nodeCursor = null;
+        }
+    }
+
+    public void releaseNodeCursor( TxNodeCursor nodeCursor )
```

Comment:
It would be nice to make this `release*Cursor` not public since the cursors will release themselves on close.